cpp/benchmarks/reshape/stack_benchmark.cu

```diff
@@ -0,0 +1,204 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <tests/utilities/column_wrapper.cuh>
+#include <tests/utilities/column_wrapper_factory.hpp>
+
+#include <cudf/reshape.hpp>
+
+#include "../fixture/benchmark_fixture.hpp"
+#include "../synchronization/synchronization.hpp"
+
+template <class T>
+class Reshape : public ::benchmark::Fixture {
+public:
+  using TypeParam = T;
+};
+
+// TODO: put it in a struct so `uniform` can be remade with different min, max
+template <typename T>
+T random_int(T min, T max)
+{
+  static unsigned seed = 13377331;
+  static std::mt19937 engine{seed};
+  static std::uniform_int_distribution<T> uniform{min, max};
+
+  return uniform(engine);
+}
+
+template <class T>
+void BM_stack(benchmark::State& state){
+  using wrapper = cudf::test::column_wrapper<T>;
+
+  const gdf_size_type num_columns{(gdf_size_type)state.range(0)};
+  const gdf_size_type column_size{(gdf_size_type)state.range(1)};
+
+  std::vector<wrapper> key_columns;
+  std::vector<wrapper> val_columns;
+
+  auto make_table = [&] (std::vector<wrapper>& cols,
+                         gdf_size_type col_size) -> cudf::table
+  {
+    cudf::test::column_wrapper_factory<T> factory;
+
+    for (gdf_size_type i = 0; i < num_columns; i++) {
+      cols.emplace_back(factory.make(col_size,
+        [=](gdf_index_type row) { return random_int(0, 10); }
+        // ,[=](gdf_index_type row) { return random_int(0, 10) < 90; }
+      ));
+    }
+
+    std::vector<gdf_column*> raw_cols(num_columns, nullptr);
+    std::transform(cols.begin(), cols.end(), raw_cols.begin(),
+                   [](wrapper &c) { return c.get(); });
+
+    return cudf::table{raw_cols.data(), num_columns};
+  };
+
+  auto key_table = make_table(key_columns, column_size);
+  // auto val_table = make_table(val_columns, column_size);
+
+  for(auto _ : state){
+    cuda_event_timer timer(state, true);
+    auto col = cudf::stack(key_table);
+    gdf_column_free(&col);
+  }
+}
+
+static void CustomArguments(benchmark::internal::Benchmark* b) {
+  for (int num_cols = 2; num_cols <= 10; num_cols*=2)
+    for (int col_size = 1000; col_size <= 10000000; col_size *= 10)
+      b->Args({num_cols, col_size});
+}
+
+// BENCHMARK_TEMPLATE_DEFINE_F(Reshape, IntStack, int64_t)
+// (::benchmark::State& state) {
+//   BM_stack<TypeParam>(state);
+// }
+
+// BENCHMARK_REGISTER_F(Reshape, IntStack)
+//   ->UseManualTime()
+//   ->Unit(benchmark::kMillisecond)
+//   ->Apply(CustomArguments);
+
+
+BENCHMARK_TEMPLATE_DEFINE_F(Reshape, StrStack, cudf::nvstring_category)
+(::benchmark::State& state) {
+  BM_stack<TypeParam>(state);
+}
+
+BENCHMARK_REGISTER_F(Reshape, StrStack)
+  ->UseManualTime()
+  ->Unit(benchmark::kMillisecond)
+  ->Args({8, 1000000});
+
+
+
+
+class String : public ::benchmark::Fixture {};
+
+void BM_mar(benchmark::State& state){
```

Comment:
mar?

Reply:
`merge_and_remap`. I was trying to benchmark the two different approaches to do `combine_column_categories`. The earlier method used repeated `merge_and_remap` across pairs of old columns and the new method is to just use `create_from_categories` on all the columns at once. The new one is faster by approx 30% on large columns and approx 50% on small ones.